src/components/windows-motion-controls.js

```diff
@@ -0,0 +1,382 @@
+    });
+
+    if (this.data.hideDisconnected) {
+      this.el.setAttribute('visible', this.controllerPresent);
+    }
+  },
+
+  play: function () {
+    this.checkIfControllerPresent();
+    this.addControllersUpdateListener();
+
+    window.addEventListener('gamepadconnected', this.checkIfControllerPresent, false);
+    window.addEventListener('gamepaddisconnected', this.checkIfControllerPresent, false);
+  },
+
+  pause: function () {
+    this.removeEventListeners();
+    this.removeControllersUpdateListener();
+
+    window.removeEventListener('gamepadconnected', this.checkIfControllerPresent, false);
+    window.removeEventListener('gamepaddisconnected', this.checkIfControllerPresent, false);
+  },
+
+  updateControllerModel: function () {
+    if (!this.data.model || this.el.getAttribute('gltf-model')) { return; }
+
+    var sourceUrl = this.createControllerModelUrl();
+    this.loadModel(sourceUrl);
+  },
+
+  /**
+   * Helper function that constructs a URL from the controller ID suffix, for future proofed
+   * art assets.
+   */
+  createControllerModelUrl: function (forceDefault) {
+    // Determine the device specific folder based on the ID suffix
+    var trackedControlsComponent = this.el.components['tracked-controls'];
+    var controller = trackedControlsComponent ? trackedControlsComponent.controller : null;
+    var device = 'default';
+    var hand = this.data.hand;
+
+    if (controller) {
+      // Read hand directly from the controller, rather than this.data, as in the case that the controller
+      // is unhanded this.data will still have 'left' or 'right' (depending on what the user inserted in to the scene).
+      // In this case, we want to load the universal model, so need to get the '' from the controller.
+      hand = controller.hand;
+
+      if (!forceDefault) {
+        var match = controller.id.match(GAMEPAD_ID_PATTERN);
+        device = ((match && match[0]) || device);
+      }
+    }
+
+    // Hand
+    var filename;
+    if (hand === 'left') filename = MODEL_LEFT_FILENAME;
+    else if (hand === 'right') filename = MODEL_RIGHT_FILENAME;
+    else filename = MODEL_UNIVERSAL_FILENAME;
+
+    // Final url
+    return MODEL_BASE_URL + device + '/' + filename;
+  },
+
+  injectTrackedControls: function () {
+    var data = this.data;
+    this.el.setAttribute('tracked-controls', {
+      idPrefix: GAMEPAD_ID_PREFIX,
+      hand: data.hand
+    });
+
+    this.updateControllerModel();
+  },
+
+  addControllersUpdateListener: function () {
+    this.el.sceneEl.addEventListener('controllersupdated', this.onControllersUpdate, false);
+  },
+
+  removeControllersUpdateListener: function () {
+    this.el.sceneEl.removeEventListener('controllersupdated', this.onControllersUpdate, false);
+  },
+
+  onControllersUpdate: function () {
+    this.checkIfControllerPresent();
+  },
+
+  onModelError: function (evt) {
+    var defaultUrl = this.createControllerModelUrl(true);
+    if (evt.detail.src !== defaultUrl) {
```

Comment:
Do we need a default model when we already have hand and left ones? If another component like `hand-controls` decides to use a custom model (hence the model property === false) is responsibility of that higher level component to handle model errors. I might be misunderstanding something though

Reply:
This logic only applies in the case where the model property === true, and is a separate concept to the default hand... our device ID's have a suffix which represents a particular hardware form factor (as exposed by the underlying Windows Mixed Reality API). 
`Spatial Controller (Spatial Interaction Source) 045E-065A`

While there is currently a single device (the Motion Controller, 045E-065A), the underlying windows API supports addition of new hardware form factors. As such, it is possible that new hardware exists (with a new ID suffix) before we upload a corresponding asset - which is why we fallback to a 'default' GLB file.

Hopefully I understand your comment correctly?

Reply:
I see. I have not seen the default model. How does it look like?

Reply:
in other cases we have used idPrefix e.g. if they will all be "Spatial Controller " but then all bets are off.  I think the version I did uses that strategy, let me find where I put that code and put it somewhere for reference

Reply:
As an alternative, what if there is an additional http request prior to model download that retrieves a manifest file with mappings of suffix -> filename, with an entry for default?

My main motivator here is that we want to avoid making changes to the core library if we need to add new, radically different, models - since people don't update library versions after releasing their apps.

Reply:
I see. About the current code approach. If new hardware is released, Is a default model better than loading the 045E-065A one? Neither of them will be representative of the new hardware. If a new released controller is radically different I expect all the mappings and buttons to be differerent. In that case, we might better have a separate new component for it. The idea of the manifest is indeed intriging. How could we make sure that the developer has control over the experience? In the demos we have done so far that work across Oculus and Vive we had to make a lot of adjustments for each of the controllers: different button number, physical distribution and capabilities (capacitive, analog, axis...), haptics, button tooltip positioning, raycasting origin, location of menus and UI that are hand locked. It worked well for us to be explicit: this is the list of controllers my app is supporting and I know their capabilities, button mapping and physical appearance. If a new controller is released I will probably want to do some integration to make sure my interaction model does not break. It might be confusing if a single component can map to multiple physical controllers / models and a dev cannot make any assumptions.

Reply:
if the functionality is same but appearance may be different, it would be better to have a default model than to throw up hands just because we're not sure exactly what this one looks like.  remember before we had real Touch controller model and we just used Vive so you could at least see what you were doing?  same for Daydream and GearVR, and Vive tracker...  That being said, we might just want to reuse existing model as the generic model.

Reply:
if functionality is actually different, then really gamepad id string should be different enough to distinguish (not just by cryptic suffix)

Reply:
I hope I'm not confusing things too much 🙂. Summing up. Based on experience, I prefer each controller component to map to one and just one physical device (with its corresponding model). As opposed to traditional games -where you don't usually see a gamepad in game- in VR the interaction model is built around the physical appearence of the controller and there are more things you want to control besides button mapping into actions.

Reply:
I'm open to evolve the current API in the future but it's probably beyond 0.7.0 scope

Reply:
yeah, I think we've been slowly evolving the controller components to use common pieces (buttons, axes, mapping, utils functions) but haven't gotten as far as model manipulations, explicit refactor may be pre-1.0 but probably not this coming release.

tactically speaking, I'd like to consider merging #2583 first, as some of the mods in this one are expanded to cover the interesting challenges the tracker poses, and rebasing this on top

Reply:
>  Based on experience, I prefer each controller component to map to one and just one physical device (with its corresponding model).

Tactically speaking, maybe it could be as simple as a generic component with the basic functionality (I had called mine `spatial-controls` along prior naming convention lines) matching on idPrefix, and then specific components that matched more specific gamepad id strings with model mappings / manipulations?

Reply:
Let's merge this first since it's higher priority than tracker support for 0.7.0

Reply:
I'd agree, this is more important, except we can't merge without the models, can we?
I think #2583 is ready to merge now.
If we have a little time, I'd rebase a fork off #2583 and PR back.  I expect the delta is not large

Reply:
I've been working on a separare PR for tracker support to implement a spectator camera. I prefer to keep https://github.com/aframevr/aframe/pull/2583 on hold and also not slow this one down.

Reply:
OK - would be happy to have this contribution from Microsoft - but if you will be doing tracker support, you may need to revisit https://github.com/aframevr/aframe/pull/3013#discussion_r136470194 -- the current changes in this PR make it impossible to match no-handedness